src/Elyon.Fastly.Api.DomainServices/InfoSessionFollowUpService.cs

```diff
@@ -49,18 +53,128 @@ public async Task SendInfoSessionFollowUpEmailAsync(InfoSessionFollowUpSpecDto s
             var organization = await _organizationsRepository
                 .GetByIdAsync(specDto.OrganizationId).ConfigureAwait(false);
 
-            if (!ValidateOrganizationForNewFollowUp(organization))
-                return;
+            if (organization.OrganizationType.Id == _smeOrganizationId)
+            {
```

Comment:
You shouldn't have such big blocks in a "if". Extract each of the "if/else" blocks into methods and try to keep the common code in here